[Uptime] 7.8 docs

justinkambic opened the pull request:
## Summary

Adds docs to 7.8. We'll forward-port this change to `master` after merging.

### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server)
- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

elasticmachine commented:
Pinging @elastic/uptime (Team:uptime)

andrewvc commented on docs/uptime/certificates.asciidoc:
```diff
@@ -0,0 +1,15 @@
+[role="screenshot"]
+image::uptime/images/certificates-page.png[Certificates]
+
+Uptime has a certificates page, which will visualize all of the TLS certificates Heartbeat
```
Our docs generally use the active voice ( I can't find a citation, but maybe @bmorelli25 has one?) . This could be written: "Use the certificates page to visualize all of your TLS certificates..."

bmorelli25 replied:
[The Elastic guide to writing style](https://wiki.elastic.co/display/MAR/Grammar%2C+punctuation%2C+and+mechanics#Grammar,punctuation,andmechanics-activevoiceActivevoice) talks about active voice.

justinkambic replied:
Please have a look at https://github.com/elastic/kibana/pull/66118/commits/24ff319d980db8ff73bacf6a182a5e3900cb4287 where I've tried to prefer active voice as much as I could. I also tried to simplify a large chunk of the copy I had added.

andrewvc commented on docs/uptime/settings.asciidoc:
```diff
@@ -25,3 +27,20 @@ to manually query Elasticsearch for data outside this pattern!
+
+=== Certificate thresholds
+
+The settings page also allows you to specify thresholds for all the TLS certificates monitored
```
Active voice: "Specify thresholds for all TLS certificates on the settings page"

andrewvc commented on docs/uptime/settings.asciidoc:
```diff
@@ -25,3 +27,20 @@ to manually query Elasticsearch for data outside this pattern!
+=== Certificate thresholds
+
+The settings page also allows you to specify thresholds for all the TLS certificates monitored
+in your environment. There is a field for maximum age, and the minimum number of days before
```
I don't think we need to be this descriptive, that part of the app is self explanatory. IMHO we shouldn't document the obvious. Would like @bmorelli25 's thoughts here.

Perhaps the content would be better if it described what the fields vs. documenting their existence. Rephrasing this as: "Use the certificate maximum age field to control the number of days before expiration Uptime should display the certificate as being in a warning state. Warnings for maximum certificate age can also be set here. Some user agents and browsers consider certificates invalid after a certain amount of time.

bmorelli25 replied:
Agreed. But looking at the screenshot, I only see "Expiration threshold" and "Age limit" as options. The docs should match those names. I'd alter Andrew's recommendation to something like this:

The settings page also allows you to specify thresholds for displaying and alerting on TLS certificates. Use the expiration threshold to specify <<STUFF>>, and the age limit field to control the <<STUFF>>. Some user agents and browsers consider certificates invalid after a certain amount of time.

andrewvc commented on docs/uptime/settings.asciidoc:
```diff
@@ -25,3 +27,20 @@ to manually query Elasticsearch for data outside this pattern!
+an `Expiration threshold` on this page. When the count of a certificate's remaining valid days falls
+below this threshold, Uptime will consider it in a warning state.
+
+The thresholds specified on this page affect how your certificates are visualized on the Certificates page,
```
This could be moved up IMHO to make the importance more clear.

bmorelli25 replied:
Yes, this should be in the first paragraph

justinkambic commented:
> Looks like there's an extra image file named Screen Shot 2020-05-11 at 2.44.25 PM.png

This was supposed to be a draft but I guess I marked it as ready 😿 

I left that as a placeholder to remind me to re-take that screenshot after https://github.com/elastic/kibana/pull/66135 was merged. I have subsequently added commits that fix this.

bmorelli25 commented on docs/uptime/settings.asciidoc:
```diff
@@ -25,3 +27,20 @@ to manually query Elasticsearch for data outside this pattern!
+in your environment. There is a field for maximum age, and the minimum number of days before
+expiration before Uptime should consider a monitor to be in warning state.
+
+For example, you may want to make sure that none of your organization's TLS certificates have been
+valid for longer than one year. This is becoming a common security requirement. Modifying the `Age limit`
```
```suggestion
For example, a common security requirement is to make sure that none of your organization's TLS certificates have been valid for longer than one year. Modifying the `Age limit`
```

justinkambic replied:
Updated in https://github.com/elastic/kibana/pull/66118/commits/4a63705487aec31499581bf529a31074dc718f4c.

justinkambic commented:
> What about something simpler

@bmorelli25 I updated the copy in https://github.com/elastic/kibana/pull/66118/commits/141c28ee33afca5fcb06e0a0f981026bb2602401. Thanks!

Feel free to comment on any more of the copy I've added, these recommendations are super helpful.

justinkambic commented on docs/uptime/alerting.asciidoc:
```diff
@@ -0,0 +1,26 @@
+[role="xpack"]
```
Changed this in https://github.com/elastic/kibana/pull/66118/commits/f0a0de9fa1de874dccb840c912d76e41359ec4fe.

justinkambic commented on docs/uptime/alerting.asciidoc:
```diff
@@ -0,0 +1,26 @@
+[role="xpack"]
+[[uptime-alerting]]
+
+== Uptime alerting
+
```
Updated in https://github.com/elastic/kibana/pull/66118/commits/04957516e37dbc9b25feb060416f355d64b57e68.

justinkambic commented:
@elasticmachine merge upstream

EamonnTP commented:
Hi @justinkambic Would it be possible to merge this PR before Wednesday? Next week, as part of the Observability docs restructuring, I plan to move the [Uptime documentation in the Kibana reference](https://www.elastic.co/guide/en/kibana/current/xpack-uptime.html) to the [Uptime guide](https://www.elastic.co/guide/en/uptime/current/index.html) and it would be great to include this new content. Thanks!

justinkambic commented:
@elasticmachine merge upstream
